### PR TITLE
Live Test config

### PR DESCRIPTION
**Pull request by g2vinay**

### CODE REVIEW

File: storage/client/tests.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+trigger: none
+
+jobs:
+    # When migrating change path to ../../eng/pipelines/templates/jobs/archetype-sdk-tests.yml
+  - template: ../../eng/pipelines/templates/jobs/archetype-sdk-tests-pre-sdk.yml
+    parameters:
+      # When migrating use ServiceDirectory
+      PomFilePath: './storage/client/pom.xml'
+      EnvVars:
+        AZURE_TEST_MODE: RECORD
```
Comment (sima-zhu) [inline storage/client/tests.yml]: AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_TENANT_ID

Are these able to get from the archetype-sdk-tests-pre-sdk.yml ?

File: storage/client/pom.xml [hunk 2]
```diff
@@ -98,7 +112,35 @@
   </dependencies>
 
   <build>
+
+    <sourceDirectory>src/main/java</sourceDirectory>
```
Comment (sima-zhu) [inline storage/client/pom.xml]: Do we exclude the sample codes for running the record mode?

File: storage/client/pom.xml [hunk 3]
```diff
@@ -98,7 +112,35 @@
...
+    <testSourceDirectory>src/test/java</testSourceDirectory>
+
     <plugins>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <compilerId>groovy-eclipse-compiler</compilerId>
```
Comment (sima-zhu) [inline storage/client/pom.xml]: I did not see these in client pom. We have ran the tests under current pom, I am wondering the groovy plugin needed to run the tests.

File: storage/client/pom.xml [hunk 4]
```diff
@@ -31,6 +31,20 @@
     <tag>HEAD</tag>
   </scm>
 
+  <pluginRepositories>
```
Comment (sima-zhu) [inline storage/client/pom.xml]: Just curious what the usage of the pluginRepository?

File: storage/client/tests.yml (hunk 1 above)
Comment (g2vinay) [reply storage/client/tests.yml]: they come from variable groups / variable in the pipeline

File: storage/client/pom.xml (hunk 3 above)
Comment (g2vinay) [reply storage/client/pom.xml]: Yeah without this mvn test will not trigger any of the tests.

File: storage/client/pom.xml (hunk 2 above)
Comment (g2vinay) [reply storage/client/pom.xml]: Sample are in the test directory, so they should get picked up with other classes.

File: storage/client/pom.xml (hunk 4 above)
Comment (g2vinay) [reply storage/client/pom.xml]: this repo, gets the dependencies for groovy-eclipse-compiler

File: storage/client/pom.xml (hunk 2 above)
Comment (sima-zhu) [reply storage/client/pom.xml]: the sample goes under src/samples/java. I did not get how I got pick up from test directory.

**Comment by g2vinay** [comment]
Added code directly to Azure:storage-proto-2.

**Pull request closed without merging**